### ISSUE TITLE
About DecoupledAttention and anchor/temp_anchor embedding

### ISSUE THREAD

**Issue by XXXVincent** [HorizonRobotics/Sparse4D#15]
Under the current released config(sparse4dv3_temporal_r50_1x8_bs6_256x704.py), **_decouple_attn = True_**, which leads to when apply **_self.graph_model_**, the **_anchor_embed_** and **_temp_anchor_embed_** will never been used. 
![image](https://github.com/HorizonRobotics/Sparse4D/assets/16273202/1953cc64-9a2a-479c-96f2-ad87d4c05838)

The multi-head-attention only uses instance_feature and temp_instance_feature as inputs. 

It seems that the explicit historical anchors is only used to get the **__temp_anchor_embed_** however **_temp_anchor_embed_** is not used in latter operations, only historical instance feature is used. 

May I ask why not use **_anchor_embed/temp_anchor_embed_**,  does this setting leads to performance drop? Is it a bad idea to collect and aggregate features nearby projected historical detections?

**Comment by linxuewu** [comment]
`query = torch.cat([query, query_pos], dim=-1)`, concatenates the anchor_embed with the instance_feature.
The anchor_embed has been used.

**Comment by XXXVincent** [comment]
Yes indeed!  Thanks! Nice work btw.
